interfaces/input_parameters/ring.py

```diff
@@ -182,182 +130,348 @@ class Ring:
-                              msg='Exactly one of '+str(syncDataTypes) +
-                              ' must be declared',
-                              exception=excpt.InputError)
-
-        # Checking that the bending_radius is passed with the bending_field
-        if bending_field is not None and bending_radius is None:
-            raise excpt.InputError("If bending_field is used, bending_radius "
-                                   + "must be defined.")
-
-        # Taking the first synchronous_data input not declared as None
-        # The assertion above ensures that only one is declared
-        for func_type, synchronous_data in zip(syncDataTypes, syncDataInput):
-            if synchronous_data is not None:
-                break
-
-        # Setting ring length, circumerence, radius, bending radius if defined
-        self.ring_length = np.array(ring_length, ndmin=1, dtype=float)
-        self.ring_circumference = np.sum(self.ring_length)
-        self.ring_radius = self.ring_circumference/(2*np.pi)
-
-        if bending_radius is not None:
-            self.bending_radius = float(bending_radius)
-        else:
-            self.bending_radius = bending_radius
+    def __init__(self, Particle, Section_list, **kwargs):
 
         # Primary particle mass and charge used for energy calculations
         # If a string is passed, will generate the relevant Particle object
         # based on the name
         if isinstance(Particle, beam.Particle):
             self.Particle = Particle
-        else:
+        elif isinstance(Particle, str):
             self.Particle = beam.make_particle(Particle)
-
-        # Reshaping the input synchronous data to the adequate format and
-        # get back the momentum program from RingOptions
-        if not isinstance(synchronous_data, dTypes._ring_program):
-            synchronous_data = \
-                dTypes._ring_program.conversions[func_type](synchronous_data)
-
-        if synchronous_data.shape[0] != len(self.ring_length):
-            raise excpt.InputDataError("ERROR in Ring: Number of sections " +
-                                       "and ring length size do not match!")
-
+        else:
+            raise excpt.InputError("The input Particle should be a " +
+                                   "Particle object or a str.")
+
+        # Getting all sections and checking their types
+        if isinstance(Section_list, RingSection):
+            Section_list = [Section_list]
+        if not isinstance(Section_list, list):
+            raise excpt.InputError("The Section_list should be a list or " +
+                                   "a single RingSection object instance.")
+        else:
+            for section in Section_list:
+                if not isinstance(section, RingSection):
+                    raise excpt.InputError(
+                        "The Section_list should be exclusively composed " +
+                        "of RingSection object instances.")
```

Comment:
No reason to require list, could also be tuple, and I think numpy array can allow array of objects as well.

I'd replace the above with:
```suggestion
if not hasattr(Section_list, '__iter__'):
    Section_list = (Section_list,)
if not all(isinstance(s, RingSection) for s in Section_list):
    raise excpt.InputError("error message")
```
Same effect but shorter and easier to read

Reply:
Included